src/variant_detection/variant_detection.cpp

```diff
@@ -70,9 +81,24 @@ void detect_junctions_in_short_reads_sam_file(std::vector<Junction> & junctions,
             seqan3::debug_stream << num_good << " good alignments from short read file." << std::endl;
         }
     }
+
+    size_t i = 0;
+    for(std::string const & ref_id : ref_ids)
+    {
+        int32_t ref_length = std::get<0>(ref_id_info[i]);
+        if (references_lengths.find(ref_id) != references_lengths.end())
```

Comment:
Here, you only check whether `ref_id` is already a key in `reference_lengths`. Shouldn't we also check whether `references_lengths[ref_id] == ref_length` and only output a warning if it is not?